druid-shell/src/platform/x11/application.rs

```diff
@@ -495,9 +495,22 @@ impl Application {
     }
 
     pub fn get_locale() -> String {
-        // TODO(x11/locales): implement Application::get_locale
-        tracing::warn!("Application::get_locale is currently unimplemented for X11 platforms. (defaulting to en-US)");
-        "en-US".into()
+        let var_non_empty = |var| match std::env::var(var) {
+            Ok(s) if s.is_empty() => None,
+            Ok(s) => Some(s),
+            Err(_) => None,
+        };
+
+        // from gettext manual
+        // https://www.gnu.org/software/gettext/manual/html_node/Locale-Environment-Variables.html#Locale-Environment-Variables
+        var_non_empty("LANGUAGE")
+            .or_else(|| var_non_empty("LC_ALL"))
+            .or_else(|| var_non_empty("LC_MESSAGES"))
+            .or_else(|| var_non_empty("LANG"))
+            // the value is priority list seperated by :
+            // See: https://www.gnu.org/software/gettext/manual/html_node/The-LANGUAGE-variable.html#The-LANGUAGE-variable
+            .and_then(|locale| locale.split(':').next().map(String::from))
+            .unwrap_or_else(|| "en-US".to_string())
```

Comment:
Is there a difference between "en-US" and "en_US"? My system uses the latter...

Reply:
I think it is better idea to `en_US`. I used `en-US` because it was used in the stub earlier.

Reply:
> The identifiers can vary in case and in the separator characters. The "-" and "_" separators are treated as equivalent, although "-" is preferred.



(from [Unicode TR #35](https://unicode.org/reports/tr35/#Identifiers))